pkg/controllers/job/job_controller.go

```diff
@@ -215,6 +215,7 @@ func NewJobController(
 	// Register actions
 	state.SyncJob = cc.syncJob
 	state.KillJob = cc.killJob
+	state.UpdateJob = cc.updateJobFn
```

Comment:
can we handle this in SyncJob?

Reply:
Good catch. I have tried and it is tricky. Maybe a separate prin future, there can be lots of flags to control different behavior.

Reply:
similar to statefulset controller, in syncJob, it just make sure pod & replicas keep sync.

Reply:
any update?